app/services/stat_service.rb

```diff
@@ -0,0 +1,40 @@
+class StatService
+
+  def initialize(params)
+    @match = params[:match]
+    @winner = params[:winner]
+    @loser = params[:loser]
```

Comment:
This part seems odd. We should probably only accept a match and let `update_stats` and `update_stats_for_match_with_new_winner` can just read the winner and loser from the match. The only time we're passing a different value than the stores winner and loser is when we're calling `update_stats_for_match_with_new_winner` and the values are always inverted, so we know what to do without being provided explicit values.